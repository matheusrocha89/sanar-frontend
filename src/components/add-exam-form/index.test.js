import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import uuid from 'uuid/v1';

import AddExamForm from './';


jest.mock('uuid/v1', () => jest.fn(() => '1234567'));


describe('AddExamForm', () => {
  it('renders correctly', () => {
    const handleClose = jest.fn();
    const addNewExam = jest.fn();
    const tree = renderer
      .create(<AddExamForm addNewExam={addNewExam} handleClose={handleClose} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with errorMessage', () => {
    const handleClose = jest.fn();
    const addNewExam = jest.fn();
    const comp = shallow(<AddExamForm addNewExam={addNewExam} handleClose={handleClose} />);
    comp.setState({ errorMessage: 'Text' });
    expect(comp.find('.AddExamForm-error-message')).toHaveLength(1);
    expect(comp.find('.AddExamForm-error-message p').text()).toEqual('Text');
  });

  it('test error message when form not valid', () => {
    const handleClose = jest.fn();
    const addNewExam = jest.fn();
    const comp = shallow(<AddExamForm addNewExam={addNewExam} handleClose={handleClose} />);
    comp.instance().validateForm();
    expect(comp.state().errorMessage).toEqual('*Campo profissão é obrigatório');
  });

  it('calls addNewExam and handleClose when form is valid', () => {
    const handleClose = jest.fn();
    const addNewExam = jest.fn();
    const event = { preventDefault: jest.fn() };
    const expectedId = '12345';

    const comp = shallow(<AddExamForm addNewExam={addNewExam} handleClose={handleClose} />);
    comp.setState({
      profession: 'abc',
      institution: 'abc',
    });
    comp.instance().onSubmit(event);

    expect(event.preventDefault).toHaveBeenCalledTimes(1);
    expect(handleClose).toHaveBeenCalledTimes(1);
    expect(addNewExam).toHaveBeenCalledWith({
      id: expectedId,
      profissao: 'abc',
      instituto: 'abc',
    });
  });
});
