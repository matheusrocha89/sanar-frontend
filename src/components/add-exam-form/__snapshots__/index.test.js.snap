// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExamForm renders correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="profession"
    >
      Profiss√£o*: 
    </label>
    <input
      className="form-control"
      id="profession"
      maxLength={70}
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor={undefined}
    >
      Instituto: 
    </label>
    <input
      className="form-control"
      id={undefined}
      maxLength={170}
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor={undefined}
    >
      Prova: 
    </label>
    <input
      className=""
      id={undefined}
      onChange={[Function]}
      type="file"
      value=""
    />
  </div>
  <div
    className="AddExamForm-buttons"
  >
    <button
      className="btn btn-success"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Cadastrar
    </button>
  </div>
</form>
`;
